src/eve-server/market/MarketMgr.cpp

```diff
@@ -298,155 +296,210 @@ void MarketMgr::InvalidateOrdersCache(uint32 regionID, uint32 typeID)
  *        return tax
  */
 
+/*
+ * Does not currently handle aurum transactions, but it would be good to do that
+ * in the future.
+ */
 bool MarketMgr::ExecuteBuyOrder(Client* seller, uint32 orderID, InventoryItemRef iRef, uint32 quantity, bool useCorp, uint32 typeID, uint32 stationID, double price, uint16 accountKey/*Account::KeyType::Cash*/) {
     Market::OrderInfo oInfo = Market::OrderInfo();
     if (!MarketDB::GetOrderInfo(orderID, oInfo)) {
         _log(MARKET__ERROR, "ExecuteBuyOrder - Failed to get order info for #%u.", orderID);
-        return true;
+
+        return false;
     }
 
-    /*  will this method also be used to buy/sell using aurm?
-     * unknown yet
-     */
+    // get buyer id and determine if buyer is player, corp, or npc/bot
+    bool isPlayer(false), isCorp(false), isTraderJoe(false), isTrader(false);
 
-    // get buyer id and determine if buyer is player or corp (or bot for later)
-    bool isPlayer(false), isCorp(false), isTrader(false);
     if (IsPlayerCorp(oInfo.ownerID)) {
-        // buyer is player corp
         isCorp = true;
     } else if (oInfo.ownerID == 1) {
         oInfo.ownerID = stDataMgr.GetOwnerID(stationID);
     } else if (IsCharacterID(oInfo.ownerID)) {
         isPlayer = true;
     } else if (IsTraderJoe(oInfo.ownerID)) {
+        isTraderJoe = true;
+    } else if (IsTrader(oInfo.ownerID)) {
```

Comment:
This function as well as `ExecuteSellOrder` have been slightly refactored and modified to support the aims of this pull request.